### PR TITLE
Update mtrie with some minor fixes and small improvements

### LINKED ISSUE #1802: [Execution state] Update mtrie with minor fixes and improvments
## Problem Definition

Several minor issues discovered while reading mtrie code.  

-  `utils.ReadShortData()` shouldn't return "rest" with extra data
- buffer size is incorrect in `flattener.EncodeStorableTrie()`, which can cause extra memory alloc 
- unnecessary error conversion code in `Ledger.Set()`
- some comments don't match code 
- some tests can be improved

## Proposed Solution

See PR opened to resolve this issue.

## Definition of Done

PR reviewed and merged with each item resolved.

### PR DESCRIPTION
**Pull request by fxamacker**
Closes #1802 

This PR updates mtrie with minor fixes and small improvements that don't break backward compatibility.

Changes include:
- Fix returned "rest" data from `utils.ReadShortData()`  (commit 2b84aaac269d1267d94f422a139d339661ddff26)
- Fix buffer size in `flattener.EncodeStorableTrie()` (commit 1fdfa14a2cc1360d47eb90c75fc35ad3777816d3)
- Remove unnecessary error conversion in `Ledger.Set()` (commit  f93cb56b2ccc290d22e8fa1f656b9ce837fe7ef0)
- Fix comments that didn't match code (commit 4fc66b904f284a3a40823507e121b7db20b18980)
- Improve test code (commit d1296e5e7ddb71405b9d9d25f0cca3549847a438)

This was done while onboarding and getting familiar with mtrie code.

### CODE REVIEW

File: ledger/complete/mtrie/forest_test.go [hunk 1]
```diff
@@ -495,14 +495,14 @@ func TestReadOrder(t *testing.T) {
 	read := &ledger.TrieRead{RootHash: testRoot, Paths: []ledger.Path{p1, p2}}
 	retPayloads, err := forest.Read(read)
 	require.NoError(t, err)
-	require.Equal(t, len(retPayloads), len(payloads))
+	require.Equal(t, len(payloads), len(retPayloads))
```
Comment (AlexHentschel) [inline ledger/complete/mtrie/forest_test.go:498]: Thanks for fixing the order of `expected` vs `actual` arguments. 

Just noticed:
I think the `expected` value should be `len(read.Paths)`:
```suggestion
	require.Equal(t, len(read.Paths), len(retPayloads))
```
My reasoning:
* we are expecting to retrieve as many payloads as we have specified paths in `read` https://github.com/onflow/flow-go/blob/f93cb56b2ccc290d22e8fa1f656b9ce837fe7ef0/ledger/complete/mtrie/forest_test.go#L495
* The revised test would work even if we had more than 2 registers in the tree

File: ledger/complete/mtrie/forest_test.go [hunk 2]
```diff
@@ -495,14 +495,14 @@ func TestReadOrder(t *testing.T) {
...
 	require.True(t, bytes.Equal(encoding.EncodePayload(retPayloads[0]), encoding.EncodePayload(payloads[0])))
 	require.True(t, bytes.Equal(encoding.EncodePayload(retPayloads[1]), encoding.EncodePayload(payloads[1])))
 
 	read = &ledger.TrieRead{RootHash: testRoot, Paths: []ledger.Path{p2, p1}}
 	retPayloads, err = forest.Read(read)
 	require.NoError(t, err)
-	require.Equal(t, len(retPayloads), len(payloads))
+	require.Equal(t, len(payloads), len(retPayloads))
```
Comment (AlexHentschel) [inline ledger/complete/mtrie/forest_test.go:505]: same here
```suggestion
	require.Equal(t, len(read.Paths), len(retPayloads))
```

File: ledger/complete/mtrie/trie/trie.go [hunk 3]
```diff
@@ -247,6 +247,7 @@ func update(
 	// in the remaining code: the registers to update are strictly larger than 1:
 	//   - either len(paths)>1
 	//   - or len(paths) == 1 and compactLeaf!= nil
+	//   - or len(paths) == 1 and parentNode != nil && !parentNode.IsLeaf()
```
Comment (AlexHentschel) [inline ledger/complete/mtrie/trie/trie.go:250]: 👍

File: ledger/complete/mtrie/trie/trie.go [hunk 4]
```diff
@@ -500,7 +501,7 @@ func (mt *MTrie) IsAValidTrie() bool {
 //
 //  For instance, if `paths` contains the following 3 paths, and bitIndex is `1`:
 //  [[0,0,1,1], [0,1,0,1], [0,0,0,1]]
-//  then `splitByPath` returns 1 and updates `paths` into:
+//  then `splitByPath` returns 2 and updates `paths` into:
```
Comment (AlexHentschel) [inline ledger/complete/mtrie/trie/trie.go:504]: 👍

**Review by AlexHentschel — APPROVED** [review]
Thanks Faye. Lots of good cleanups. 👍

File: ledger/partial/ledger.go [hunk 5]
```diff
@@ -113,27 +113,7 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, trieUpdate *
...
-				path := paths[i]
-				pathToKey[path] = key
-			}
-
-			keys := make([]ledger.Key, 0, len(pErr.Paths))
-			for _, path := range pErr.Paths {
-				keys = append(keys, pathToKey[path])
-			}
-			return ledger.DummyState, nil, &ledger.ErrMissingKeys{Keys: keys}
-		}
```
Comment (ramtinms) [inline ledger/partial/ledger.go:136]: I think we need this for the challenges created by VNs

Comment (ramtinms) [reply ledger/partial/ledger.go:136]: hmm, thats odd, it suppose to be some on verification integration tests

Comment (ramtinms) [reply ledger/partial/ledger.go:136]: it seems all tests are passing so we can drop it :D

**Review by ramtinms — APPROVED** [review]

**Pull request merged**